irohad/torii/impl/command_service_transport_grpc.cpp

```diff
@@ -239,6 +253,12 @@ namespace torii {
             return false;
           }
 
+          if (last_tx_status_received) {
+            // force stream to end because no more tx statuses will arrive.
+            // it is thread safe because of synchronization on current_thread
+            return false;
+          }
+
           log_->debug("status written, {}", client_id);
```

Comment:
I get the following order in a minimal example:
```
a next
a complete
a finally
b next
take while
complete
```
where `a` is status bus, and `b` is consensus events observable.
It means that it will execute `take_while` after `finally`, and this log `status written` should be moved before `last_tx_status_received` conditional.